File: apps/ios/Barns/BarnsApp/Domain/UseCases/Care/CompleteCareTaskUseCase.swift
/// Marks a care task as completed in local state.
struct CompleteCareTaskUseCase {
    private let repository: CareRepository

    init(repository: CareRepository) {
        self.repository = repository
    }

    func execute(id: CareTask.ID) async throws {
        try await repository.completeCareTask(id: id)
    }
}

File: apps/android/app/src/main/java/com/barns/app/domain/usecase/care/CompleteCareTaskUseCase.kt
package com.barns.app.domain.usecase.care

import com.barns.app.domain.repository.CareRepository

/**
 * Marks a care task as completed in local state.
 */
class CompleteCareTaskUseCase(
    private val repository: CareRepository,
) {
    suspend fun execute(id: String) = repository.completeCareTask(id)
}
